### ISSUE TITLE
Add a Google Maps / OpenMapData GPS geocoding lens?

### ISSUE THREAD

**Issue by okennedy** [UBOdin/mimir#111]
e.g., : https://ubodin.slack.com/files/okennedy/F2JNN91LL/corner_lookup.rb

**Comment by Legacy25** [comment]
I dabbled with the google maps api for a toy project. Here is a snippet of code demonstrating how we can invoke the api to do a geocoding request. Maybe it will be useful to someone trying to do this -

Disclaimer - Might be out of date...
```
package controllers.geodata

import java.sql.Timestamp
import java.util.Calendar

import play.api.libs.ws.WS
import play.api.Play

import scala.concurrent.ExecutionContext.Implicits.global
import play.api.Play.current
import scala.concurrent.duration.Duration
import scala.concurrent.{Await, Future}

object Geocode {

  val googleGeocodeApiUrl = "https://maps.googleapis.com/maps/api/geocode/json"

  def getLatLong(streetAddress: String): (String, Double, Double, Timestamp) = {
    val future = WS.url(googleGeocodeApiUrl)
      .withQueryString(
        "address" -> streetAddress,
        "key" -> Play.current.configuration.getString("googleServerKey").get
      ).get().map{

      wsResponse => (wsResponse.json \ "status").as[String] match {
        case "OK" => {
          val firstResult = (wsResponse.json \ "results")(0)
          val formattedAddress = (firstResult \\ "formatted_address").head.as[String]
          val lat = (firstResult \\ "lat").head.as[Double]
          val long = (firstResult \\ "lng").head.as[Double]
          val ts = new Timestamp(Calendar.getInstance().getTime.getTime)

          (formattedAddress, lat, long, ts)
        }
//        case "ZERO_RESULTS" =>
//        case "OVER_QUERY_LIMIT" =>
//        case "REQUEST_DENIED" =>
//        case "INVALID_REQUEST" =>
//        case "UNKNOWN_ERROR" =>
        case _ => throw new Exception("Geocoding Failed!")
      }
    }

    Await.result(future, Duration(5, "seconds"))
  }
}
```

**Comment by okennedy** [comment]
Nice.  I'd swap in a full path to the lat/lng... but this should be trivial to throw onto a lens.

**Comment by okennedy** [comment]
If Google blocks this off or prevents access, another possible tool might be based on OSM.  This just showed up on HN today: https://github.com/maddevsio/ariadna

**Comment by mrb24** [comment]
Added by commit 0d4df401b2efde02380e56c7f89b583d355b8685
Merged with master by commit 6d2662121c75a7f9846b33e30150e630d1f41075
